### PR TITLE
Allow type "file:..." to enable creation of SBOMs from tar and other package formats

### PR DESCRIPTION
**Pull request by malt3**
Signed-off-by: Malte Poll <mp@edgeless.systems>

### CODE REVIEW

**Review by kzantow — COMMENTED** [review]
Hi @malt3, thanks for the submission! I think to be consistent we should probably add a `file` parameter, which is mutually exclusive to `image` and `path` instead of having the `path` double as both `dir` and `file`. Would this work for you?

**Comment by malt3** [comment]
Sure thing. It sounds a little bit confusing to have `path` which resolves to `dir:` and `file` which resolves to `file:` but renaming the existing parameter is probably even worse as it is breaking change.
What do you think about keeping `path` -> `dir:` for backwards compat and adding  both `file` -> `file:` and `dir` -> `dir:` as parameters?
Just a suggestion. I can implement this any way you like :)

File: README.md [hunk 1]
```diff
@@ -58,6 +58,16 @@ Use the `path` parameter, relative to the repository root:
     path: ./build/
 ```
 
+### Scan a specific file
```
Comment (kzantow) [inline README.md:61]: This looks good but could you also update the parameters section below?

**Comment by kzantow** [comment]
@malt3 thanks for the contribution -- this looks good, I just need to get a successful run of the tests. I'm not sure why they haven't run yet... It appears GH may be having some issues with actions at the moment. I'll try to get them kicked off later today.

**Comment by felickz** [comment]
This PR was pushed with 0.13.0 but not seeing the file property being observed: https://github.com/anchore/sbom-action/issues/383#issuecomment-1295362506

**Pull request merged**